### PR TITLE
Initial version of glossary. Closes #197

### LINKED ISSUE #197: Add key terms to the glossary in the reference section
Comments from the initial editorial checks carried out as part of the peer review process (see https://github.com/carpentries-lab/reviews/issues/15) highlight that key terms need to be added to the glossary (see the [Reference](https://carpentries-incubator.github.io/docker-introduction/reference.html) section).

The editor highlights that terms and their definitions can be added here or, where terms are already provided in the Carpentries [Glosario](https://glosario.carpentries.org/) glossary, the term can be added and linked to Glosario.

### PR DESCRIPTION
**Pull request by aturner-epcc**
Initial set of glossary terms

### CODE REVIEW

File: reference.md [hunk 1]
```diff
@@ -4,6 +4,57 @@ layout: reference
...
+   <dt>Operating system (OS)</dt>
+   <dd><a href="https://glosario.carpentries.org/en/#operating_system" alt="Link to Carpentries Glossario">See the Carpentries Glossario</a></dd>
+   <dt>Reproducible research</dt>
+   <dd><a href="https://glosario.carpentries.org/en/#reproducible_research" alt="Link to Carpentries Glossario">See the Carpentries Glossario</a></dd>
+   <dt>Software library</dt>
+   <dd><a href="https://glosario.carpentries.org/en/#library" alt="Link to Carpentries Glossario">See the Carpentries Glossario</a></dd>
+   <dt>Tar archive</dt>
+   <dd>A file archive format commonly used in Unix-like operating systems that combines multiple files into a single file. tar archive files are used as the export format of <em>Docker images</em>.</dd>
+   <dt>Virtualization</dt>
+   <dd><em>Containers</em> are an example of virtualization – having a second “virtual” computer running and accessible from a <em>host computer</em>.</dd>
```
Comment (jcohen02) [inline reference.md:57]: I was thinking whether there is a way to explain this in relation to something that provides a virtual, software-based representation of hardware that can support running artefacts designed to run on the "emulated" hardware? I think maybe the description you've provided is more straightforward and I'm perhaps thinking about this in the context of a more technical description which I guess we don't need.

**Review by jcohen02 — APPROVED** [review]
Many thanks for putting this together @aturner-epcc, it's looking great. 🙂 
I've provided a few comments - approving the PR and will leave it with you to decide how you want to proceed in terms of the suggestions.

File: reference.md [hunk 2]
```diff
@@ -4,6 +4,57 @@ layout: reference
 
 ## Glossary
 
-FIXME
+<dl>
+   <dt>Command-line argument/option</dt>
+   <dd><a href="https://glosario.carpentries.org/en/#command_line_argument" alt="Link to Carpentries Glossario">See the Carpentries Glossario</a></dd>
```
[the remark this answers is not in the document]
Comment (sstevens2) [reply reference.md:9]: Maybe we could say something like all definitions are from the Carpentries Glossario unless otherwise specified?

**Comment by sstevens2** [comment]
pinging @aturner-epcc to see if you want to make any updates from @jcohen02's comments or would like to merge?

File: reference.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (aturner-epcc) [reply reference.md:9]: I think @jcohen02 suggestion is the correct one, I will update the PR

File: reference.md (hunk 1 above)
Comment (aturner-epcc) [reply reference.md:57]: I think this is a simple glossary so we do not need "complete" technical descriptions. I lean towards leaving the simple definition for now but if you really feel strongly we can add something more detailed

**Review by jcohen02 — APPROVED** [review]
Thanks @aturner-epcc, looks good to me - have run locally via Jekyll and everything seems to render correctly. We could perhaps expand or modify some of the definitions here but then it might get too wordy - I guess this is the challenge of trying to explain complex concepts very concisely. That might be something to consider but this looks great for now.
